velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -133,6 +135,59 @@ struct TimestampWithTimezoneSupport {
 
 } // namespace
 
+template <typename T>
+struct WeekFunction : public InitSessionTimezone<T>,
+                      public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE int64_t getWeek(const std::tm& time) {
+    // The computation of ISO week from date follows the algorithm here:
+    // https://en.wikipedia.org/wiki/ISO_week_date
+    int64_t week = floor(
+                       10 + (time.tm_yday + 1) -
+                       (time.tm_wday ? time.tm_wday : kDaysInWeek)) /
+        kDaysInWeek;
+
+    if (week == 0) {
```

Comment:
Please add the comments like those from here https://cplusplus.com/forum/general/220155/ as it makes the code more understandable.

Reply:
Updated, thanks for the suggestion!